src/couch/src/couch_db_updater.erl

```diff
@@ -1467,6 +1474,16 @@ make_doc_summary(#db{compression = Comp}, {Body0, Atts0}) ->
     SummaryBin = ?term_to_bin({Body, Atts}),
     couch_file:assemble_file_chunk(SummaryBin, couch_crypto:hash(md5, SummaryBin)).
 
+
+get_meta_body_size(Meta, Summary) ->
+    case lists:keyfind(ejson_size, 1, Meta) of
+        {ejson_size, ExternalSize} ->
+            ExternalSize;
+        false ->
+            couch_compress:decompress(Summary)
```

Comment:
This is a bug right? You should be doing `?term_size(couch_compress:decompress(Summary))`.